### PR TITLE
Implement DXR object collections

### PR DESCRIPTION
**Pull request by HansKristian-Work**
Fix https://github.com/HansKristian-Work/vkd3d-proton/issues/568
Fix https://github.com/HansKristian-Work/vkd3d-proton/issues/592

### CODE REVIEW

File: libs/vkd3d/raytracing_pipeline.c [hunk 1]
```diff
@@ -110,14 +137,7 @@ static ULONG d3d12_state_object_release(struct d3d12_state_object *state_object)
     TRACE("%p decreasing refcount to %u.\n", state_object, refcount);
 
     if (!refcount)
-    {
```
Comment (doitsujin) [inline libs/vkd3d/raytracing_pipeline.c:112]: Can there ever be a situation where the app can re-claim a reference to the object after the (public) ref count has reached zero? In that case, `AddRef` would have to handle that case and add a private ref again.

File: libs/vkd3d/device.c [hunk 2]
```diff
@@ -4750,9 +4754,12 @@ static void d3d12_device_caps_init_shader_model(struct d3d12_device *device)
          * FP16, Denorm modes (float controls extension)
          */
         if (device->device_info.float16_int8_features.shaderFloat16 &&
-            device->device_info.subgroup_extended_types_features.shaderSubgroupExtendedTypes)
+                device->device_info.storage_16bit_features.storageBuffer16BitAccess &&
+                device->device_info.subgroup_extended_types_features.shaderSubgroupExtendedTypes)
```
Comment (doitsujin) [inline libs/vkd3d/device.c:4758]: Do we have enough 16-bit support in dxil-spirv to enable `Native16BitShaderOpsSupported`?

**Review by doitsujin — COMMENTED** [review]
LGTM, just two questions.

File: libs/vkd3d/device.c (hunk 2 above)
Comment (HansKristian-Work) [reply libs/vkd3d/device.c:4758]: Yes, I think so.

File: libs/vkd3d/raytracing_pipeline.c (hunk 1 above)
Comment (HansKristian-Work) [reply libs/vkd3d/raytracing_pipeline.c:112]: I doubt it. The DXR wording says: "Collection to include in state object. Enclosing state object holds a ref on the existing collection."
I proved it's not a real reference, but I don't think this means being able to make those objects live again.

**Review by doitsujin — APPROVED** [review]

File: libs/vkd3d/device.c (hunk 2 above)
Comment (doitsujin) [reply libs/vkd3d/device.c:4758]: Guess we should write a test and enable that at some point then, but can be done separately.

**Pull request merged**